src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckCollector.java

```diff
@@ -46,35 +44,37 @@
 // Created this class as a parent because of the ObjectFactory bean that is used with Collectors which destroys the bean after use. These services need to be destroyed after usage.
 public abstract class BlackDuckCollector extends MessageContentCollector {
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    private final Optional<BlackDuckBucketService> bucketService;
-    private final Optional<BlackDuckService> blackDuckService;
+    private final BlackDuckServicesFactory blackDuckServicesFactory;
+    private final BlackDuckBucketService bucketService;
     private final BlackDuckBucket blackDuckBucket;
+    private final boolean isValidServer;
 
     public BlackDuckCollector(final JsonExtractor jsonExtractor, final List<MessageContentProcessor> messageContentProcessorList, final Collection<ProviderContentType> contentTypes, final BlackDuckProperties blackDuckProperties) {
         super(jsonExtractor, messageContentProcessorList, contentTypes);
 
-        final Optional<BlackDuckServicesFactory> blackDuckServicesFactory = blackDuckProperties.createBlackDuckHttpClientAndLogErrors(logger)
-                                                                                .map(blackDuckHttpClient -> blackDuckProperties.createBlackDuckServicesFactory(blackDuckHttpClient, new Slf4jIntLogger(logger)));
-        blackDuckService = blackDuckServicesFactory.map(BlackDuckServicesFactory::createBlackDuckService);
-        bucketService = blackDuckServicesFactory.map(BlackDuckServicesFactory::createBlackDuckBucketService);
+        final Optional<BlackDuckServicesFactory> blackDuckServicesFactoryOptional = blackDuckProperties.createBlackDuckHttpClientAndLogErrors(logger)
+                                                                                        .map(blackDuckHttpClient -> blackDuckProperties.createBlackDuckServicesFactory(blackDuckHttpClient, new Slf4jIntLogger(logger)));
+        isValidServer = blackDuckServicesFactoryOptional.isPresent();
+        blackDuckServicesFactory = blackDuckServicesFactoryOptional.orElse(null);
+        bucketService = blackDuckServicesFactoryOptional.map(BlackDuckServicesFactory::createBlackDuckBucketService).orElse(null);
         blackDuckBucket = new BlackDuckBucket();
     }
 
     public Optional<String> getProjectComponentQueryLink(final String projectVersionUrl, final String link, final String componentName) {
         final Optional<String> projectLink = getProjectLink(projectVersionUrl, link);
-        return projectLink.flatMap(optionalProjectLink -> getProjectComponentQueryLink(optionalProjectLink, componentName));
+        return projectLink.map(optionalProjectLink -> getProjectComponentQueryLink(optionalProjectLink, componentName));
```

Comment:
rename optionalProjectLink to projectLink, because here it isn't optional inside the lambda.